crates/dojo-core/src/executor.cairo

```diff
@@ -1,25 +1,58 @@
+use core::serde::Serde;
 #[contract]
 mod Executor {
     use array::{ArrayTrait, ArrayTCloneImpl};
+    use serde::Serde;
     use clone::Clone;
+    use box::BoxTrait;
     use traits::Into;
+    use dojo_core::execution_context::Context;
+    use dojo_core::integer::u250;
+    use dojo_core::interfaces::IWorldDispatcher;
+    use dojo_core::auth::components::AuthRole;
     use starknet::contract_address::ContractAddressIntoFelt252;
+    use starknet::{get_caller_address, get_tx_info};
 
     use dojo_core::serde::SpanSerde;
 
     const EXECUTE_ENTRYPOINT: felt252 =
         0x0240060cdb34fcc260f41eac7474ee1d7c80b7e3607daff9ac67c7ea2ebb1c44;
 
+    /// Executes a System by calling its execute entrypoint.
+    ///
+    /// # Arguments
+    ///
+    /// * `class_hash` - The class hash of the System to execute.
+    /// * `role_id` - The role ID for System to assume during the execution.
+    /// * `calldata` - The calldata to pass to the System.
+    ///
+    /// # Returns
+    ///
+    /// The return value of the System's execute entrypoint.
     #[external]
-    fn execute(class_hash: starknet::ClassHash, calldata: Span<felt252>) -> Span<felt252> {
-        let world_address = starknet::get_caller_address();
+    fn execute(
+        class_hash: starknet::ClassHash, role_id: u250, calldata: Span<felt252>
+    ) -> Span<felt252> {
```

Comment:
Passing of `role_id` when calling `execute` in order to provide the assumed role of the System during execution

Reply:
What does this look like for the default scoped roles? Is there a way to have it assume the default role? Maybe if it is passed as `0`?

Reply:
Yeah that's a good point! Maybe for now we can set `0` to assume default role or we could do `Option<role_id>` and do some matching, open to both approaches 🙂

As for the default scoped roles, I would say we will have to put the component in the context somehow so we can retrieve the scoped role and have the system assume it